### ISSUE TITLE
training loss

### ISSUE THREAD

**Issue by 1286710929** [dk-liang/CLTR#3]
when we reprodece your code, we find that the training loss could not reduce fast as your reported result. And the result of sha is 70.385 and 114.081. There is a large gap compared to your reported results.:

2022-09-17 01:55:57,093 - CLTR - INFO - begin test
2022-09-17 01:56:37,525 - CLTR - INFO - Testing Epoch:[1240/1500]	 mae=73.967	 mse=118.639	 best_mae=71.473	
2022-09-17 01:56:44,394 - CLTR - INFO - Training Epoch:[1241/1500]	 loss=3.62841	 lr=0.000100	 epoch_time=6.869
2022-09-17 01:56:51,514 - CLTR - INFO - Training Epoch:[1242/1500]	 loss=3.46821	 lr=0.000100	 epoch_time=7.118
2022-09-17 01:56:58,223 - CLTR - INFO - Training Epoch:[1243/1500]	 loss=3.55540	 lr=0.000100	 epoch_time=6.708
2022-09-17 01:57:05,025 - CLTR - INFO - Training Epoch:[1244/1500]	 loss=3.51130	 lr=0.000100	 epoch_time=6.801
2022-09-17 01:57:12,066 - CLTR - INFO - Training Epoch:[1245/1500]	 loss=3.39875	 lr=0.000100	 epoch_time=7.040
2022-09-17 01:57:19,134 - CLTR - INFO - Training Epoch:[1246/1500]	 loss=3.46755	 lr=0.000100	 epoch_time=7.067
2022-09-17 01:57:26,227 - CLTR - INFO - Training Epoch:[1247/1500]	 loss=3.53449	 lr=0.000100	 epoch_time=7.092
2022-09-17 01:57:32,900 - CLTR - INFO - Training Epoch:[1248/1500]	 loss=3.61158	 lr=0.000100	 epoch_time=6.672
2022-09-17 01:57:39,863 - CLTR - INFO - Training Epoch:[1249/1500]	 loss=3.47778	 lr=0.000100	 epoch_time=6.962
2022-09-17 01:57:46,597 - CLTR - INFO - Training Epoch:[1250/1500]	 loss=3.47975	 lr=0.000100	 epoch_time=6.733
2022-09-17 01:57:53,532 - CLTR - INFO - Training Epoch:[1251/1500]	 loss=3.43447	 lr=0.000100	 epoch_time=6.934
2022-09-17 01:58:00,620 - CLTR - INFO - Training Epoch:[1252/1500]	 loss=3.51294	 lr=0.000100	 epoch_time=7.087
2022-09-17 01:58:07,619 - CLTR - INFO - Training Epoch:[1253/1500]	 loss=3.42724	 lr=0.000100	 epoch_time=6.997
2022-09-17 01:58:14,715 - CLTR - INFO - Training Epoch:[1254/1500]	 loss=3.52697	 lr=0.000100	 epoch_time=7.095
2022-09-17 01:58:21,246 - CLTR - INFO - Training Epoch:[1255/1500]	 loss=3.68258	 lr=0.000100	 epoch_time=6.530
2022-09-17 01:58:28,036 - CLTR - INFO - Training Epoch:[1256/1500]	 loss=3.47060	 lr=0.000100	 epoch_time=6.789
2022-09-17 01:58:35,009 - CLTR - INFO - Training Epoch:[1257/1500]	 loss=3.52460	 lr=0.000100	 epoch_time=6.972
2022-09-17 01:58:41,836 - CLTR - INFO - Training Epoch:[1258/1500]	 loss=3.50146	 lr=0.000100	 epoch_time=6.826
2022-09-17 01:58:48,671 - CLTR - INFO - Training Epoch:[1259/1500]	 loss=3.58137	 lr=0.000100	 epoch_time=6.834
2022-09-17 01:58:55,733 - CLTR - INFO - Training Epoch:[1260/1500]	 loss=3.44624	 lr=0.000100	 epoch_time=7.061
2022-09-17 01:58:55,734 - CLTR - INFO - begin test
2022-09-17 01:59:26,869 - CLTR - INFO - Testing Epoch:[1260/1500]	 mae=70.385	 mse=114.081	 best_mae=70.385	
2022-09-17 01:59:33,723 - CLTR - INFO - Training Epoch:[1261/1500]	 loss=3.57292	 lr=0.000100	 epoch_time=6.854
2022-09-17 01:59:40,602 - CLTR - INFO - Training Epoch:[1262/1500]	 loss=3.52274	 lr=0.000100	 epoch_time=6.878
2022-09-17 01:59:47,400 - CLTR - INFO - Training Epoch:[1263/1500]	 loss=3.48122	 lr=0.000100	 epoch_time=6.797
2022-09-17 01:59:54,421 - CLTR - INFO - Training Epoch:[1264/1500]	 loss=3.45710	 lr=0.000100	 epoch_time=7.020
2022-09-17 02:00:01,293 - CLTR - INFO - Training Epoch:[1265/1500]	 loss=3.53121	 lr=0.000100	 epoch_time=6.871
2022-09-17 02:00:08,249 - CLTR - INFO - Training Epoch:[1266/1500]	 loss=3.53630	 lr=0.000100	 epoch_time=6.955
2022-09-17 02:00:15,238 - CLTR - INFO - Training Epoch:[1267/1500]	 loss=3.41089	 lr=0.000100	 epoch_time=6.988
2022-09-17 02:00:21,989 - CLTR - INFO - Training Epoch:[1268/1500]	 loss=3.55521	 lr=0.000100	 epoch_time=6.750
2022-09-17 02:00:28,751 - CLTR - INFO - Training Epoch:[1269/1500]	 loss=3.63312	 lr=0.000100	 epoch_time=6.761
2022-09-17 02:00:35,698 - CLTR - INFO - Training Epoch:[1270/1500]	 loss=3.41506	 lr=0.000100	 epoch_time=6.945
2022-09-17 02:00:42,523 - CLTR - INFO - Training Epoch:[1271/1500]	 loss=3.55783	 lr=0.000100	 epoch_time=6.824
2022-09-17 02:00:49,606 - CLTR - INFO - Training Epoch:[1272/1500]	 loss=3.44602	 lr=0.000100	 epoch_time=7.082
2022-09-17 02:00:56,604 - CLTR - INFO - Training Epoch:[1273/1500]	 loss=3.51345	 lr=0.000100	 epoch_time=6.997
2022-09-17 02:01:03,575 - CLTR - INFO - Training Epoch:[1274/1500]	 loss=3.42050	 lr=0.000100	 epoch_time=6.969
2022-09-17 02:01:10,444 - CLTR - INFO - Training Epoch:[1275/1500]	 loss=3.48487	 lr=0.000100	 epoch_time=6.868
2022-09-17 02:01:17,235 - CLTR - INFO - Training Epoch:[1276/1500]	 loss=3.48882	 lr=0.000100	 epoch_time=6.790
2022-09-17 02:01:24,114 - CLTR - INFO - Training Epoch:[1277/1500]	 loss=3.57055	 lr=0.000100	 epoch_time=6.878
2022-09-17 02:01:31,314 - CLTR - INFO - Training Epoch:[1278/1500]	 loss=3.45037	 lr=0.000100	 epoch_time=7.199
2022-09-17 02:01:38,042 - CLTR - INFO - Training Epoch:[1279/1500]	 loss=3.68802	 lr=0.000100	 epoch_time=6.726
2022-09-17 02:01:44,786 - CLTR - INFO - Training Epoch:[1280/1500]	 loss=3.51901	 lr=0.000100	 epoch_time=6.743
2022-09-17 02:01:44,787 - CLTR - INFO - begin test
2022-09-17 02:02:12,237 - CLTR - INFO - Testing Epoch:[1280/1500]	 mae=75.275	 mse=123.192	 best_mae=70.385	
2022-09-17 02:02:19,001 - CLTR - INFO - Training Epoch:[1281/1500]	 loss=3.62225	 lr=0.000100	 epoch_time=6.763
2022-09-17 02:02:25,931 - CLTR - INFO - Training Epoch:[1282/1500]	 loss=3.52498	 lr=0.000100	 epoch_time=6.929
2022-09-17 02:02:32,735 - CLTR - INFO - Training Epoch:[1283/1500]	 loss=3.51817	 lr=0.000100	 epoch_time=6.803
2022-09-17 02:02:39,512 - CLTR - INFO - Training Epoch:[1284/1500]	 loss=3.59665	 lr=0.000100	 epoch_time=6.776
2022-09-17 02:02:46,403 - CLTR - INFO - Training Epoch:[1285/1500]	 loss=3.57462	 lr=0.000100	 epoch_time=6.889
2022-09-17 02:02:53,304 - CLTR - INFO - Training Epoch:[1286/1500]	 loss=3.45410	 lr=0.000100	 epoch_time=6.901
2022-09-17 02:03:00,091 - CLTR - INFO - Training Epoch:[1287/1500]	 loss=3.52542	 lr=0.000100	 epoch_time=6.786
2022-09-17 02:03:06,966 - CLTR - INFO - Training Epoch:[1288/1500]	 loss=3.43454	 lr=0.000100	 epoch_time=6.873
2022-09-17 02:03:14,036 - CLTR - INFO - Training Epoch:[1289/1500]	 loss=3.47092	 lr=0.000100	 epoch_time=7.069
2022-09-17 02:03:20,975 - CLTR - INFO - Training Epoch:[1290/1500]	 loss=3.50124	 lr=0.000100	 epoch_time=6.939
2022-09-17 02:03:27,941 - CLTR - INFO - Training Epoch:[1291/1500]	 loss=3.43352	 lr=0.000100	 epoch_time=6.964
2022-09-17 02:03:35,042 - CLTR - INFO - Training Epoch:[1292/1500]	 loss=3.39298	 lr=0.000100	 epoch_time=7.100
2022-09-17 02:03:42,024 - CLTR - INFO - Training Epoch:[1293/1500]	 loss=3.50791	 lr=0.000100	 epoch_time=6.981
2022-09-17 02:03:48,689 - CLTR - INFO - Training Epoch:[1294/1500]	 loss=3.48784	 lr=0.000100	 epoch_time=6.664
2022-09-17 02:03:55,622 - CLTR - INFO - Training Epoch:[1295/1500]	 loss=3.52300	 lr=0.000100	 epoch_time=6.932
2022-09-17 02:04:02,859 - CLTR - INFO - Training Epoch:[1296/1500]	 loss=3.28457	 lr=0.000100	 epoch_time=7.236
2022-09-17 02:04:09,545 - CLTR - INFO - Training Epoch:[1297/1500]	 loss=3.57269	 lr=0.000100	 epoch_time=6.685
2022-09-17 02:04:16,391 - CLTR - INFO - Training Epoch:[1298/1500]	 loss=3.59400	 lr=0.000100	 epoch_time=6.846
2022-09-17 02:04:23,223 - CLTR - INFO - Training Epoch:[1299/1500]	 loss=3.54231	 lr=0.000100	 epoch_time=6.831
2022-09-17 02:04:30,576 - CLTR - INFO - Training Epoch:[1300/1500]	 loss=3.40182	 lr=0.000100	 epoch_time=7.352
2022-09-17 02:04:30,577 - CLTR - INFO - begin test

**Comment by dk-liang** [comment]
We have no plan to release the config of the ShanghaiTech Dataset since I am busy with other projects. You can try to train the NWPU and JHU dataset

**Comment by 1286710929** [comment]
> We have no plan to release the config of the ShanghaiTech Dataset since I am busy with other projects. You can try to train the NWPU and JHU dataset

Thank you for your reply! However, when we train the JHU dataset, the training loss still could not reduce so fast as your reported NWPU log.

**Comment by dk-liang** [comment]
> > We have no plan to release the config of the ShanghaiTech Dataset since I am busy with other projects. You can try to train the NWPU and JHU dataset
> 
> Thank you for your reply! However, when we train the JHU dataset, the training loss still could not reduce so fast as your reported NWPU log.

Hi, how many GPUs do you use? The experiments are conducted on 4 x GPUs

**Comment by 1286710929** [comment]
> 
Due to the device restrictions, we use one 3090 GPU.

**Comment by 1286710929** [comment]
> > > We have no plan to release the config of the ShanghaiTech Dataset since I am busy with other projects. You can try to train the NWPU and JHU dataset
> > 
> > 
> > Thank you for your reply! However, when we train the JHU dataset, the training loss still could not reduce so fast as your reported NWPU log.
> 
> Hi, how many GPUs do you use? The experiments are conducted on 4 x GPUs

May 4 x GPUs would perform much better than one GPU?

**Comment by dk-liang** [comment]
> > > > We have no plan to release the config of the ShanghaiTech Dataset since I am busy with other projects. You can try to train the NWPU and JHU dataset
> > > 
> > > 
> > > Thank you for your reply! However, when we train the JHU dataset, the training loss still could not reduce so fast as your reported NWPU log.
> > 
> > 
> > Hi, how many GPUs do you use? The experiments are conducted on 4 x GPUs
> 
> May 4 x GPUs would perform much better than one GPU?

It is possible. Maybe you need to change some hyper-parameters (e.g., lr, query_number).
